### PR TITLE
IT-3400 - ripple works if button is changing it's size. 

### PR DESCRIPTION
**Pull request by damianmef**
It's not working always, but for small changes (to +35%) yes

### CODE REVIEW

File: src/ripple.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var Ripple = {
         // Default values.
         var props = {
             event: 'mousedown',
-            transition: 600
+            transition: 950
```
Comment (kvas-damian) [inline src/ripple.js:7]: it's a default value for a prop. Did you check if we don't set it anywhere in the code?

File: src/ripple.js [hunk 2]
```diff
@@ -78,8 +78,8 @@ var Ripple = {
             // No need to set positioning because ripple should be child of target and to it's relative position.
             // rippleContainer.style.left    = left + (((window.pageXOffset || document.scrollLeft) - (document.clientLeft || 0)) || 0) + "px";
             // rippleContainer.style.top     = top + (((window.pageYOffset || document.scrollTop) - (document.clientTop || 0)) || 0) + "px";
-            rippleContainer.style.width   = width + "px";
-            rippleContainer.style.height  = height + "px";
+            rippleContainer.style.width   = "100%";
+            rippleContainer.style.height  = height*1.2 + "px";
```
Comment (kvas-damian) [inline src/ripple.js]: why do we have to increase `height`? Isn't it safer to use set `height` in percent value of parent?

Comment (kvas-damian) [reply src/ripple.js]: BTW, because of this change `ripple` is going out of parent component boundary

File: src/ripple.js [hunk 3]
```diff
@@ -78,8 +78,8 @@ var Ripple = {
             // No need to set positioning because ripple should be child of target and to it's relative position.
             // rippleContainer.style.left    = left + (((window.pageXOffset || document.scrollLeft) - (document.clientLeft || 0)) || 0) + "px";
             // rippleContainer.style.top     = top + (((window.pageYOffset || document.scrollTop) - (document.clientTop || 0)) || 0) + "px";
-            rippleContainer.style.width   = width + "px";
-            rippleContainer.style.height  = height + "px";
+            rippleContainer.style.width   = "100%";
```
Comment (kvas-damian) [inline src/ripple.js]: does it work ok when parent element has `border`?

File: src/ripple.js (hunk 1 above)
Comment (damianmef) [reply src/ripple.js:7]: We do not overwrite.

File: src/ripple.js (hunk 2 above)
Comment (damianmef) [reply src/ripple.js]: Ok, it's safer to don't touch this property, I will revert it.

File: src/ripple.js (hunk 3 above)
Comment (damianmef) [reply src/ripple.js]: Good catch. I added it to this width calculation. There will be a problem if element will have different border sizes, but original calculation also does not take this into account:
```
var targetBorder = parseInt((getComputedStyle(target).borderWidth).replace('px', ''));
```

File: src/ripple.js [hunk 4]
```diff
@@ -78,7 +78,7 @@ var Ripple = {
             // No need to set positioning because ripple should be child of target and to it's relative position.
             // rippleContainer.style.left    = left + (((window.pageXOffset || document.scrollLeft) - (document.clientLeft || 0)) || 0) + "px";
             // rippleContainer.style.top     = top + (((window.pageYOffset || document.scrollTop) - (document.clientTop || 0)) || 0) + "px";
-            rippleContainer.style.width   = width + "px";
+            rippleContainer.style.width   = "calc(100% + " + border * 2 + "px)";
```
Comment (kvas-damian) [inline src/ripple.js:81]: it's ok for now, but in future it can break we use different `border-width` on alef and right ;)

**Review by kvas-damian — APPROVED** [review]

File: src/ripple.js (hunk 4 above)
Comment (kvas-damian) [reply src/ripple.js:81]: you already mentioned that in https://github.com/bethinkpl/vue-ripple-directive/pull/2#discussion_r1011351652 :+1:

File: src/ripple.js [hunk 5]
```diff
@@ -78,7 +78,7 @@ var Ripple = {
             // No need to set positioning because ripple should be child of target and to it's relative position.
             // rippleContainer.style.left    = left + (((window.pageXOffset || document.scrollLeft) - (document.clientLeft || 0)) || 0) + "px";
             // rippleContainer.style.top     = top + (((window.pageYOffset || document.scrollTop) - (document.clientTop || 0)) || 0) + "px";
-            rippleContainer.style.width   = width + "px";
```
Comment (kvas-damian) [inline src/ripple.js:81]: variable `width` is not used now

Comment (damianmef) [reply src/ripple.js:81]: It's used:
```
maxX        = Math.max(dx, width - dx),
```

**Pull request merged**